modules/calib3d/src/usac/homography_solver.cpp

```diff
@@ -0,0 +1,347 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "../precomp.hpp"
+#include "../usac.hpp"
+#ifdef HAVE_EIGEN
+#include <Eigen/Eigen>
+#endif
+
+namespace cv { namespace usac {
+class HomographyMinimalSolver4ptsGEMImpl : public HomographyMinimalSolver4ptsGEM {
+private:
+    const Mat * points_mat;
+    const float * const points;
+public:
+    explicit HomographyMinimalSolver4ptsGEMImpl (const Mat &points_) :
+        points_mat(&points_), points ((float*) points_.data) {}
+
+    int estimate (const std::vector<int>& sample, std::vector<Mat> &models) const override {
+        /*
+          IMPORTANT: READ BEFORE DOWNLOADING, COPYING, INSTALLING OR USING.
+
+          By downloading, copying, installing or using the software you agree to this license.
```

Comment:
License headers including external references should be placed in the file header (before any code)

Reply:
agree. @ivashmak, it would be better to move full-scale copyright notices into the header files. In .cpp files it's up to you. You can just reference the header files.

Reply:
moved this license to usac.hpp